src/login/logind-session.c

```diff
@@ -1393,7 +1393,11 @@ int session_kill(Session *s, KillWhom whom, int signo) {
         if (!s->scope)
                 return -ESRCH;
 
-        return manager_kill_unit(s->manager, s->scope, whom, signo, NULL);
+        switch (whom) {
+                case KILL_ALL: return manager_kill_unit(s->manager, s->scope, KILL_ALL, signo, NULL);
+                case KILL_LEADER: return pidref_kill(&s->leader, signo);
+                default: return -EINVAL;
+        }
```

Comment:
```suggestion
        switch (whom) {
                case KILL_ALL:
                        return manager_kill_unit(s->manager, s->scope, KILL_ALL, signo, NULL);
                        
                case KILL_LEADER:
                        return pidref_kill(&s->leader, signo);
                        
                default:
                        assert_not_reached();
        }
```

Reply:
Done.

Reply:
(sorry, the comment has been edited. Could you update to the new revision too?)

Reply:
Sorry, what should I update ?

Reply:
Oh nvm I think I got it.